src/textual/widgets/tabs.py

```diff
@@ -0,0 +1,350 @@
+                label_cell_cursor,
+                label_cell_cursor + len_label_text,
+            )
+            label_cell_cursor += len_label_text + rpad
+
+            # Cache the position of the whole tab, i.e. the range that can be clicked
+            self._selection_range_cache[tab.name] = (
+                label_cell_cursor - lpad,
+                label_cell_cursor + len_label_text + rpad,
+            )
+
+    def get_underline_bar(self, console: Console) -> RenderResult:
+        """Yields the bar that appears below the tab labels in the Tabs widget"""
+        if self.tabs:
+            ranges = self._label_range_cache
+            tab_index = int(self.bar_offset)
+            next_tab_index = (tab_index + 1) % len(ranges)
+            range_values = list(ranges.values())
+            tab1_start, tab1_end = range_values[tab_index]
+            tab2_start, tab2_end = range_values[next_tab_index]
+
+            bar_start = tab1_start + (tab2_start - tab1_start) * (
+                self.bar_offset - tab_index
+            )
+            bar_end = tab1_end + (tab2_end - tab1_end) * (self.bar_offset - tab_index)
+        else:
+            bar_start = 0
+            bar_end = 0
+        underline = UnderlineBar(
+            highlight_range=(bar_start, bar_end),
+            highlight_style=self.active_bar_style,
+            background_style=self.inactive_bar_style,
+            clickable_ranges=self._selection_range_cache,
+        )
+        yield from console.render(underline)
+
+
+class Tabs(Widget):
+    """Widget which displays a set of horizontal tabs.
+
+    Args:
+        tabs (list[Tab]): A list of Tab objects defining the tabs which should be rendered.
+        active_tab (str, optional): The name of the tab that should be active on first render.
+        active_tab_style (StyleType): Style to apply to the label of the active tab.
+        active_bar_style (StyleType): Style to apply to the underline of the active tab.
+        inactive_tab_style (StyleType): Style to apply to the label of inactive tabs.
+        inactive_bar_style (StyleType): Style to apply to the underline of inactive tabs.
+        inactive_text_opacity (float): Opacity of the text labels of inactive tabs.
+        animation_duration (float): The duration of the tab change animation, in seconds.
+        animation_function (str): The easing function to use for the tab change animation.
+        tab_padding (int, optional): The padding at the side of each tab. If None, tabs will
+            automatically be padded such that they fit the  available horizontal space.
+        search_by_first_character (bool): If True, entering a character on your keyboard
+            will activate the next tab (in left-to-right order) with a label starting with
+            that character.
+    """
+
+    DEFAULT_STYLES = "height: 2;"
+
+    _active_tab_name: Reactive[str | None] = Reactive("")
+    _bar_offset: Reactive[float] = Reactive(0.0)
```

Comment:
I *think* you can leave out the type, and Mypy will assume it is the default.

i.e. this works, and Mypy will still assume it is a floar:

```python
_bar_offset = Reactive(0.0)
```

Just letting you know. I don't have a problem with the repetition.